presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java

```diff
@@ -156,11 +160,23 @@ else if (VARBINARY.equals(type)) {
         else if (DATE.equals(type)) {
             // convert to midnight in default time zone
             long utcMillis = DAYS.toMillis(type.getLong(block, position));
-            long localMillis = getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
+            long localMillis = UTC.getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
             statement.setDate(parameter, new Date(localMillis));
         }
+        else if (TIMESTAMP.equals(type)) {
+            // convert to default time zone
+            long utcMillis = type.getLong(block, position);
+            long localMillis = UTC.getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
```

Comment:
This most probably conflicts (logically) with #9385. I'd rather that this part of the PR is extracted as separate and merged after #9385 is merged.

Reply:
Also, since Postgres `timestamp` data type seems not to follow the standard, we may need to mention that in the docs.

Reply:
Good point. I will change the last commit to only support varbinary.